src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.js

```diff
@@ -37,15 +39,28 @@ function MentionUserRenderer(props) {
     let accountID;
     let displayNameOrLogin;
     let navigationRoute;
+    const tnode = cloneDeep(props.tnode);
+    const getMentionDisplayText = (displayText, accountId, userLogin = '') => {
+        if (accountId && userLogin !== displayText) {
+            return displayText;
+        }
+        if (!LoginUtils.areEmailsFromSamePrivateDomain(displayText, props.currentUserPersonalDetails.login)) {
+            return displayText;
+        }
+
+        return displayText.split('@')[0];
+    };
 
     if (!_.isEmpty(htmlAttribAccountID)) {
         const user = lodashGet(personalDetails, htmlAttribAccountID);
         accountID = parseInt(htmlAttribAccountID, 10);
         displayNameOrLogin = LocalePhoneNumber.formatPhoneNumber(lodashGet(user, 'login', '')) || lodashGet(user, 'displayName', '') || translate('common.hidden');
+        displayNameOrLogin = getMentionDisplayText(displayNameOrLogin, htmlAttribAccountID, lodashGet(user, 'login', ''));
         navigationRoute = ROUTES.PROFILE.getRoute(htmlAttribAccountID);
-    } else if (!_.isEmpty(props.tnode.data)) {
+    } else if (!_.isEmpty(tnode.data)) {
         // We need to remove the LTR unicode and leading @ from data as it is not part of the login
-        displayNameOrLogin = props.tnode.data.replace(CONST.UNICODE.LTR, '').slice(1);
+        displayNameOrLogin = tnode.data.replace(CONST.UNICODE.LTR, '').slice(1);
+        tnode.data = tnode.data.replace(displayNameOrLogin, getMentionDisplayText(displayNameOrLogin, htmlAttribAccountID));
```

Comment:
NAB: I know you didn't introduce this variable but generally we prefer non-abbreviated available names
```suggestion
        tnode.data = tnode.data.replace(displayNameOrLogin, getMentionDisplayText(displayNameOrLogin, htmlAttributeAccountID));
```

Reply:
Also a comment for why we replace tnode.data would be good